libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -589,7 +607,17 @@ void AP_Airspeed::update_calibration(uint8_t i, float raw_pressure)
             GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Airspeed %u unhealthy", i + 1);
         } else {
             GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Airspeed %u calibrated", i + 1);
-            param[i].offset.set_and_save(state[i].cal.sum / state[i].cal.count);
+            float calibrated_offset = state[i].cal.sum / state[i].cal.count;
+            // check if new offset differs too greatly from last calibration, indicating pitot uncovered in wind
+            if (fixed_wing_parameters != nullptr) {
+                float airspeed_min = fixed_wing_parameters->airspeed_min.get();
+                // use percentage of ARSPD_FBW_MIN as criteria for max allowed change in offset
+                float max_change = 0.5*(sq((1 + (max_speed_pcnt * 0.01))*airspeed_min) - sq(airspeed_min));
+                if (max_speed_pcnt > 0 && (abs(calibrated_offset-param[i].offset) > max_change)) {
+                    GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Arspd offset change large;cover and recal");
```

Comment:
We might want to just accept a larger then 50 character status text. It would be more readable, and it's actually worth printing the sensor ID out that triggered this, if a user didn't make a mistake covering the tube and it's actually a failed sensor it helps them diagnose them. The logs would obviously have the same data, but given that we don't get them all the time...

Reply:
done